backends/ent/annotations.go

```diff
@@ -27,7 +28,8 @@ func (backend *Backend) createAnnotations(data ...*ent.Annotation) error {
 		builder := tx.Annotation.Create().
 			SetDocumentID(data[idx].DocumentID).
 			SetName(data[idx].Name).
-			SetValue(data[idx].Value)
+			SetValue(data[idx].Value).
+			SetIsUnique(data[idx].IsUnique)
```

Comment:
Is this missing a query to check if an annotation with `IsUnique` and the same name already exists?

Reply:
Uniqueness is enforced by this unique index in the schema:

```go
		index.Fields("document_id", "name").
			Unique().
			Annotations(entsql.IndexWhere("is_unique = true")).
			StorageKey("idx_document_unique_annotations"),
```

and a few lines down from here it runs

```go
	err = tx.Annotation.CreateBulk(builders...).
		OnConflict().
		UpdateNewValues().
		Exec(backend.ctx)
```

so if it hits that unique constraint it updates the existing annotation for that document with the new value

Reply:
Ah perfect, thanks I missed that!